libsyclinterface/source/dpctl_sycl_kernel_bundle_interface.cpp

```diff
@@ -0,0 +1,784 @@
+
+            kernel interop_kernel = make_kernel<cl_be>(ocl_kernel_from_kb, ctx);
+
+            return wrap(new kernel(interop_kernel));
+        } catch (std::exception const &e) {
+            error_handler(e, __FILE__, __func__, __LINE__);
+            return nullptr;
+        }
+    }
+    else {
+        error_handler("Kernel " + std::string(kernel_name) + " not found.",
+                      __FILE__, __func__, __LINE__);
+        return nullptr;
+    }
+}
+
+#ifdef DPCTL_ENABLE_L0_PROGRAM_CREATION
+
+#ifdef __linux__
+static const char *zeLoaderName = DPCTL_LIBZE_LOADER_FILENAME;
+static const int zeLibLoadFlags = RTLD_NOLOAD | RTLD_NOW | RTLD_LOCAL;
+#elif defined(_WIN64)
+static const char *zeLoaderName = "ze_loader.dll";
+static const int zeLibLoadFlags = 0;
+#else
+#error "Level Zero program compilation is unavailable for this platform"
+#endif
+
+constexpr sycl::backend ze_be = sycl::backend::ext_oneapi_level_zero;
+
+struct ze_loader
+{
+public:
+    static ze_loader &get()
+    {
+        static ze_loader _loader;
+        return _loader;
+    }
+
+    template <typename retTy> retTy getSymbol(const char *name)
+    {
+        if (!opened) {
+            error_handler("The Level-Zero loader dynamic library could not "
+                          "be opened.",
+                          __FILE__, __func__, __LINE__);
+
+            return nullptr;
+        }
+        return zeLib.getSymbol<retTy>(name);
+    }
+
+private:
+    dpctl::DynamicLibHelper zeLib;
+    bool opened;
+    ze_loader() : zeLib(zeLoaderName, zeLibLoadFlags), opened(zeLib.opened()) {}
+};
+
+typedef ze_result_t (*zeModuleCreateFT)(ze_context_handle_t,
+                                        ze_device_handle_t,
+                                        const ze_module_desc_t *,
+                                        ze_module_handle_t *,
+                                        ze_module_build_log_handle_t *);
+const char *zeModuleCreate_Name = "zeModuleCreate";
+zeModuleCreateFT get_zeModuleCreate()
+{
+    static auto st_zeModuleCreateF =
+        ze_loader::get().getSymbol<zeModuleCreateFT>(zeModuleCreate_Name);
+
+    return st_zeModuleCreateF;
+}
+
+typedef ze_result_t (*zeModuleDestroyFT)(ze_module_handle_t);
+const char *zeModuleDestroy_Name = "zeModuleDestroy";
+zeModuleDestroyFT get_zeModuleDestroy()
+{
+    static auto st_zeModuleDestroyF =
+        ze_loader::get().getSymbol<zeModuleDestroyFT>(zeModuleDestroy_Name);
+
+    return st_zeModuleDestroyF;
+}
+
+typedef ze_result_t (*zeKernelCreateFT)(ze_module_handle_t,
+                                        const ze_kernel_desc_t *,
+                                        ze_kernel_handle_t *);
+const char *zeKernelCreate_Name = "zeKernelCreate";
+zeKernelCreateFT get_zeKernelCreate()
+{
+    static auto st_zeKernelCreateF =
+        ze_loader::get().getSymbol<zeKernelCreateFT>(zeKernelCreate_Name);
+
+    return st_zeKernelCreateF;
+}
+
+std::string _GetErrorCode_ze_impl(ze_result_t code)
```

Comment:
Same here. Not all `ze_result_t` codes are checked. Only those that used Level-Zero functions are documented to return. Perhaps using `switch` is better than chaining `if`/`else`.